iron/gpg.c

```diff
@@ -51,17 +51,73 @@
 
 
 
-static int      gpg_now;        //  Everything that timestamps a packet during the same "transaction" should
-                                //  use this time, so they all get timestamped the same.
-static char   * user_login;     //  Stores the login of the user running the process. Set at initialization.
+static int      gpg_now;                    //  Everything that timestamps a packet during the same "transaction"
+                                            //  should use this time, so they all get timestamped the same
 
-static char   * user_ssh_dir;   //  The ~/.ssh directory for user_login.
+static char   * user_login = NULL;          //  Stores login of user running process. Set at initialization
 
-static char   * user_ironcore_dir;  //  The ~/.ssh/ironcore directory for user_login.
+static char   * user_ssh_dir = NULL;        //  The $(user_dir)/.ssh/ directory for user_login
+static char   * user_ironcore_dir = NULL;   //  The $(user_ssh_dir)/ironcore/ directory for user_login
 
-static char   * hostname = "";  //  Name of the host to which ironsftp connected
+static char   * hostname = "";              //  Name of the host to which ironsftp connected
 
-static int      inited = 0;     //  Indicates that the process has initialized everything needed for IronSFTP
+static int      inited = 0;                 //  Indicates that process has initialized everything needed for IronSFTP
+
+
+/**
+ *  Set user paths
+ *
+ *  Given a directory path that should include the desired user name (i.e /tmp/ironssh/regress/pokey),
+ *  set up the user login and the paths to the directory containing the user's RSA key and the directory where the
+ *  new ironcore keys are stored. For the example path, sets the user_login to pokey, the user_ssh_dir to
+ *  <path>/.ssh, and the user_ironcore_dir to <path>/.ssh/ironcore.
+ *
+ *  This is intended primarily for testing purposes.
+ *
+ *  @param path The base path for the user
+ */
+static void
+iron_set_user_dir(const char * path)
```

Comment:
Cut and paste error. `iron_set_user` sets the login, then calls `iron_set_user_path` to set the paths. I'll fix the comment.